pkg/shell/base_index.js

```diff
@@ -205,12 +205,16 @@ function Frames(index, setupIdleResetTimers) {
         if (new_frame) {
             list[component] = frame;
             document.getElementById("content").appendChild(frame);
-            // Set background to dark mode
-            if (cockpit.dark_mode && frame.contentDocument && frame.contentDocument.documentElement) {
-                // --pf-global--BackgroundColor--light-200
-                const dark_mode_background = '#0f1214';
-                frame.contentDocument.documentElement.style.background = dark_mode_background;
-            }
+
+            frame.contentWindow.addEventListener("readystatechange", function(event) {
+                if (!event.target.documentElement)
+                    return;
+                if (index.dark_mode) {
+                    event.target.documentElement.classList.add('pf-theme-dark');
+                } else {
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17914-20221201-104014-14c78ccb-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)